Patch status: limbo, needs improvement to be upstreamed

No execinfo with musl, and execinfo.h usage was guarded in [1],
but then stack_trace.cc does not check this. We can lazily re-use
the __UCLIBC__ macros for that purpose until improved though.

[1] https://crrev.com/599f4045d195c1bc1fa5bff34124d2d3d95c317a
--- a/src/3rdparty/chromium/base/debug/stack_trace.cc
+++ b/src/3rdparty/chromium/base/debug/stack_trace.cc
@@ -259,5 +259,5 @@
   // symbol_level=0.
   return false;
-#elif defined(__UCLIBC__) || defined(_AIX)
+#elif !defined(__GLIBC__) || defined(_AIX)
   // StackTrace::OutputToStream() is not implemented under uclibc, nor AIX.
   // See https://crbug.com/706728
@@ -300,4 +300,5 @@
 }
 
+#if defined(__GLIBC__) && !defined(_AIX)
 void StackTrace::OutputToStream(std::ostream* os) const {
   OutputToStreamWithPrefix(os, {});
@@ -314,4 +315,5 @@
   OutputToStreamWithPrefixImpl(os, prefix_string);
 }
+#endif
 
 std::string StackTrace::ToString() const {
@@ -321,5 +323,5 @@
 std::string StackTrace::ToStringWithPrefix(cstring_view prefix_string) const {
   std::stringstream stream;
-#if !defined(__UCLIBC__) && !defined(_AIX)
+#if defined(__GLIBC__) && !defined(_AIX)
   OutputToStreamWithPrefix(&stream, prefix_string);
 #endif
@@ -345,5 +347,5 @@
 
 std::ostream& operator<<(std::ostream& os, const StackTrace& s) {
-#if !defined(__UCLIBC__) && !defined(_AIX)
+#if defined(__GLIBC__) && !defined(_AIX)
   s.OutputToStream(&os);
 #else
--- a/src/3rdparty/chromium/base/debug/stack_trace.h
+++ b/src/3rdparty/chromium/base/debug/stack_trace.h
@@ -119,5 +119,5 @@
   void PrintWithPrefix(cstring_view prefix_string) const;
 
-#if !defined(__UCLIBC__) && !defined(_AIX)
+#if defined(__GLIBC__) && !defined(_AIX)
   // Resolves backtrace to symbols and write to stream.
   void OutputToStream(std::ostream* os) const;
